### PR TITLE
console: handle dbs with no public schema

### PR DESCRIPTION
**Pull request by beerose**
Solves #2652 

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Console

### Solution and Design

* Fetch all schemas names,
* If there's no `public` schema, then set `currentSchema` as the first one from the list,
* If there's no `public` schema, I'm showing info message `No public schema` (to be discussed).

### Steps to test and verify
Open the console with no `public` schema. Click on `data`. You should be redirected to the first schema from the list.

### CODE REVIEW

**Comment by rikinsk** [comment]
Regarding the message we can say, "No public schema found, showing <schema_name> instead"

**Comment by beerose** [comment]
I handled redirects to `public` and I also changed logic in `add trigger` where `public` schema was hardcoded as default.

**Comment by rikinsk** [comment]
Also, we can now allow deleting public schema from the console

File: CHANGELOG.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ Read more about the session argument for computed fields in the [docs](https://h
 - console: fix underscores not being displayed on raw sql page (close #4754) (#4799)
 - console: fix visiting view modify page overwriting raw sql content (fix #4798) (#4810)
 - console: add help button and move about page to settings (#4848)
+- console: handle databases with no public schema (#4358) (#2652)
```
Comment (beerose) [inline CHANGELOG.md:84]:
```suggestion
- console: handle databases with no public schema (#4358) (fix #2652)
```

File: console/src/components/Services/Data/DataActions.js [hunk 2]
```diff
@@ -459,7 +459,18 @@ const fetchFunctionInit = (schema = null) => (dispatch, getState) => {
...
 
-const updateCurrentSchema = (schemaName, redirect = true) => dispatch => {
+const updateCurrentSchema = (
+  schemaName,
+  redirect = true,
+  schemaList = []
+) => dispatch => {
+  if (
+    schemaList.length &&
+    !schemaList.map(s => s.schema_name).includes(schemaName)
```
Comment (soorajshankar) [inline console/src/components/Services/Data/DataActions.js:469]: can we simplify this a bit? this is constructing an extra array and full array iteration is happening in all scenarios.
try this 
```jsx
schemaList.find(s=>s.schema_name===schemaName)!==undefined
```

File: console/src/components/Services/Data/DataPageContainer.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ const DataPageContainer = ({
   const styles = require('../../Common/TableCommon/Table.scss');
 
   if (!schemaList.map(s => s.schema_name).includes(currentSchema)) {
```
Comment (soorajshankar) [inline console/src/components/Services/Data/DataPageContainer.js:24]: same here ^

File: console/src/components/Services/Data/DataRouter.js [hunk 4]
```diff
@@ -145,57 +141,64 @@ const dataRouterUtils = (connect, store, composeOnEnterHooks) => {
...
-    let currentSchema = nextState.params.schema;
-    if (
-      currentSchema === null ||
-      currentSchema === undefined ||
-      currentSchema === ''
-    ) {
-      currentSchema = 'public';
-    }
+      if (!currentSchema) {
+        if (schemaList.map(s => s.schema_name).includes('public')) {
```
Comment (soorajshankar) [inline console/src/components/Services/Data/DataRouter.js:155]: same here^

File: console/src/components/Services/EventTrigger/Add/AddTrigger.js [hunk 5]
```diff
@@ -43,12 +43,13 @@ import {
 class AddTrigger extends Component {
   constructor(props) {
     super(props);
-    this.props.dispatch(loadTableList('public'));
+    this.props.dispatch(loadTableList(props.currentSchema));
```
Comment (soorajshankar) [inline console/src/components/Services/EventTrigger/Add/AddTrigger.js:46]: Dispatching from a constructor is a bit unsafe, can we move this to lifecycle methods?

Comment (soorajshankar) [reply console/src/components/Services/EventTrigger/Add/AddTrigger.js:46]: This was there in the code so far, 
🙈Do as you wish.

Comment (beerose) [reply console/src/components/Services/EventTrigger/Add/AddTrigger.js:46]: It was changed in Scheduled Triggers PR.

**Comment by beerose** [comment]
This will be a part of: https://github.com/hasura/graphql-engine/pull/5654

**Pull request closed without merging**